### ISSUE TITLE
debug_traceBlockByHash - Erigon shows TXN Reverted Error not seen in Geth

### ISSUE THREAD

**Issue by nhovsmith-cb** [ledgerwatch/erigon#9677]
#### Describe the bug
When making calls to `debug_traceBlockByHash`, the erigon client returns inconsistent results to that of geth. In particular, it displays some transactions as having errors and being reverted with different gas fee. In contrast, geth shows them as successful. 

#### System information

Erigon version: `./erigon --version`
"erigon/2.58.1"

OS & Version: Windows/Linux/OSX
linux-amd64/go1.20.7

Chain/Network: Ethereum Sepolia

#### Expected behaviour
[gethResponse.json](https://github.com/ledgerwatch/erigon/files/14561117/gethResponse.json)
Note in the last txn returned from `debug_traceBlockByHash` the txn does not appear to be reverted:
```
{
            "txHash": "0xaa771b89ffe370a73561cb2759293e1d763e091438244493bdda7184a6cd02b4",
            "result": {
                "from": "0xb2248390842d3c4acf1d8a893954afc0eac586e5",
                "gas": "0x30a84",
                "gasUsed": "0x23e43",
                "to": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                "input": "0x3e5aa08200000000000000000000000000000000000000000000000000000000000157ad0000000000000000000000000000000000000000000000000000000000079ed70000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000122abbb000000000000000000000000000000000000000000000000000000000122aeaa",
                "calls": [
                    {
                        "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                        "gas": "0x21d85",
                        "gasUsed": "0x159b6",
                        "to": "0x02ee6e3f7dd2a02eb608d47f9dcebfe209ac61fc",
                        "input": "0x3e5aa08200000000000000000000000000000000000000000000000000000000000157ad0000000000000000000000000000000000000000000000000000000000079ed70000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000122abbb000000000000000000000000000000000000000000000000000000000122aeaa",
                        "calls": [
                            {
                                "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                                "gas": "0x20727",
                                "gasUsed": "0x211",
                                "to": "0xf6134c5849fe8177163747288d41283b271b1624",
                                "input": "0xe83a2d82",
                                "output": "0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000006018244d5e6a55b9bdd7c7e1dc9403a2b4483a6f718cee420d7d7c8cf3e2ee40801cf2c7000db810d6f05d4161bc3919d61b97862822d38a28dad03a4c6259d1f016ec4666e6c8f1da16cc0f52d855d113cf5d97e07dfade9fc08af1ca6b1392d011bf4c1536ed1bb7bd5d15748ca227e70674fd61f026fbd2216fac43bdfe5cf01237d894e5859f897d9980fe49a9f9e56fbc82d3ed4aebdd676ba5a2fb96a3601a02eca1909fea99f505a127bd4d215c23152a5e0e20c6209275247f2168f02",
                                "type": "STATICCALL"
                            },
                            {
                                "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                                "gas": "0x1f8c4",
                                "gasUsed": "0x60",
                                "to": "0xf6134c5849fe8177163747288d41283b271b1624",
                                "input": "0x1f6d6ef7",
                                "output": "0x00000000000000000000000000000000000000000000000000000006574a002c",
                                "type": "STATICCALL"
                            },
                            {
                                "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                                "gas": "0x1ed61",
                                "gasUsed": "0xe65",
                                "to": "0x38f918d0e9f1b721edaa41302e399fa1b79333a9",
                                "input": "0xeca067ad",
                                "output": "0x0000000000000000000000000000000000000000000000000000000000079ed9",
                                "calls": [
                                    {
                                        "from": "0x38f918d0e9f1b721edaa41302e399fa1b79333a9",
                                        "gas": "0x1d914",
                                        "gasUsed": "0x171",
                                        "to": "0x6d010a6a9c66e1c1fb2db5f2317858b4ee9dbd39",
                                        "input": "0xeca067ad",
                                        "output": "0x0000000000000000000000000000000000000000000000000000000000079ed9",
                                        "value": "0x0",
                                        "type": "DELEGATECALL"
                                    }
                                ],
                                "type": "STATICCALL"
                            },
                            {
                                "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                                "gas": "0x1dc8f",
                                "gasUsed": "0x7068",
                                "to": "0x38f918d0e9f1b721edaa41302e399fa1b79333a9",
                                "input": "0x86598a567e5f578f301059614191b7f05c44070c537ce645708194b1940c92c2e485311c0000000000000000000000000000000000000000000000000000000000079ed7000000000000000000000000000000000000000000000000000000000122abbb000000000000000000000000000000000000000000000000000000000122aeaa",
                                "output": "0x00000000000000000000000000000000000000000000000000000000000157ada25e27c0521f1a62089c19b8e971a1015cd8b977f8b1d6a5845f273a20dfeca46e25af56ea0b2e167a0446753db4ca90e61036dc46fac42193653f607009299b9a71bb22b0c244e81c09429d1270448c155661db6e9f32088b3ab9f6dc3601b2",
                                "calls": [
                                    {
                                        "from": "0x38f918d0e9f1b721edaa41302e399fa1b79333a9",
                                        "gas": "0x1d211",
                                        "gasUsed": "0x6d1d",
                                        "to": "0x6d010a6a9c66e1c1fb2db5f2317858b4ee9dbd39",
                                        "input": "0x86598a567e5f578f301059614191b7f05c44070c537ce645708194b1940c92c2e485311c0000000000000000000000000000000000000000000000000000000000079ed7000000000000000000000000000000000000000000000000000000000122abbb000000000000000000000000000000000000000000000000000000000122aeaa",
                                        "output": "0x00000000000000000000000000000000000000000000000000000000000157ada25e27c0521f1a62089c19b8e971a1015cd8b977f8b1d6a5845f273a20dfeca46e25af56ea0b2e167a0446753db4ca90e61036dc46fac42193653f607009299b9a71bb22b0c244e81c09429d1270448c155661db6e9f32088b3ab9f6dc3601b2",
                                        "value": "0x0",
                                        "type": "DELEGATECALL"
                                    }
                                ],
                                "value": "0x0",
                                "type": "CALL"
                            },
                            {
                                "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                                "gas": "0x14d9f",
                                "gasUsed": "0x8037",
                                "to": "0x38f918d0e9f1b721edaa41302e399fa1b79333a9",
                                "input": "0x7a88b107000000000000000000000000b2248390842d3c4acf1d8a893954afc0eac586e592772b148795479003814bb2ebe718a9b8c42f4c8ef699c21b7ccc2b7a6ecff3",
                                "output": "0x0000000000000000000000000000000000000000000000000000000000079ed9",
                                "calls": [
                                    {
                                        "from": "0x38f918d0e9f1b721edaa41302e399fa1b79333a9",
                                        "gas": "0x14568",
                                        "gasUsed": "0x7d01",
                                        "to": "0x6d010a6a9c66e1c1fb2db5f2317858b4ee9dbd39",
                                        "input": "0x7a88b107000000000000000000000000b2248390842d3c4acf1d8a893954afc0eac586e592772b148795479003814bb2ebe718a9b8c42f4c8ef699c21b7ccc2b7a6ecff3",
                                        "output": "0x0000000000000000000000000000000000000000000000000000000000079ed9",
                                        "value": "0x0",
                                        "type": "DELEGATECALL"
                                    }
                                ],
                                "value": "0x0",
                                "type": "CALL"
                            }
                        ],
                        "value": "0x0",
                        "type": "DELEGATECALL"
                    }
                ],
                "value": "0x0",
                "type": "CALL"
            }
        }
    ]
}
```

#### Actual behaviour
[erigonResponse.json](https://github.com/ledgerwatch/erigon/files/14560846/erigonResponse.json)
Note in the last txn returned by `debug_traceBlockByHash` from an Erigon client, the execution appears as reverted and has different gas values from the expected result: 

```
{
            "txHash": "0xaa771b89ffe370a73561cb2759293e1d763e091438244493bdda7184a6cd02b4",
            "result": {
                "from": "0xb2248390842d3c4acf1d8a893954afc0eac586e5",
                "gas": "0x30a84",
                "gasUsed": "0xf5bf",
                "to": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                "input": "0x3e5aa08200000000000000000000000000000000000000000000000000000000000157ad0000000000000000000000000000000000000000000000000000000000079ed70000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000122abbb000000000000000000000000000000000000000000000000000000000122aeaa",
                "output": "0x3cd27eb6",
                "error": "execution reverted",
                "calls": [
                    {
                        "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                        "gas": "0x21d85",
                        "gasUsed": "0x112e",
                        "to": "0x02ee6e3f7dd2a02eb608d47f9dcebfe209ac61fc",
                        "input": "0x3e5aa08200000000000000000000000000000000000000000000000000000000000157ad0000000000000000000000000000000000000000000000000000000000079ed70000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000122abbb000000000000000000000000000000000000000000000000000000000122aeaa",
                        "output": "0x3cd27eb6",
                        "error": "execution reverted",
                        "calls": [
                            {
                                "from": "0x6c97864ce4bef387de0b3310a44230f7e3f1be0d",
                                "gas": "0x20727",
                                "gasUsed": "0x8b",
                                "to": "0xf6134c5849fe8177163747288d41283b271b1624",
                                "input": "0xe83a2d82",
                                "output": "0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000",
                                "type": "STATICCALL"
                            }
                        ],
                        "type": "DELEGATECALL"
                    }
                ],
                "value": "0x0",
                "type": "CALL"
            }
        }
    ]
}
```

#### Steps to reproduce the behaviour
Send the following request to a geth and erigon node 
```
curl --location ENDPOINT \
--header 'Content-Type: application/json' \
--data '{"jsonrpc":"2.0","id":1,"method":"debug_traceBlockByHash","params":["0x9b34a045c6f32c5190d164e2372fe6efa4254b4b098168b26b9a913e7bb4f443",{"timeout":"60s","tracer":"callTracer"}]}'
```
Diffing the results will highlight the issue with inconsistent responses (don't use a json diff). Plz fix ASAP

**Comment by indanielo** [comment]
Similar issue here with debug_traceBlockByNumber for block: 19434486 -> if you check transactionHash = "0x24525505af95825a084c2166035d3437a0db77fde2c930f31b213722d144ee96" -> erigon is only returning 1 internal call with an error. However, the reality is that the transaction was success with 6 internal calls.

**Comment by DmytroShalaiev** [comment]
Same issue

**Comment by zhaozewei** [comment]
Same issue

**Comment by VladLypovyi** [comment]
fixed in [v2.59.0](https://github.com/ledgerwatch/erigon/releases/tag/v2.59.0) release
